src/coreclr/vm/olevariant.h

```diff
@@ -624,14 +265,10 @@ class OleVariant
     static void ClearVariantArray(void* oleArray, SIZE_T cElements, MethodTable* pInterfaceMT, PCODE pManagedMarshalerCode);
 
 #ifdef FEATURE_COMINTEROP
-    static void MarshalArrayVariantOleToCom(VARIANT* pOleVariant, VariantData* pComVariant);
-    static void MarshalArrayVariantComToOle(VariantData* pComVariant, VARIANT* pOleVariant);
-    static void MarshalArrayVariantOleRefToCom(VARIANT* pOleVariant, VariantData* pComVariant);
+    static void MarshalArrayVariantOleToObject(const VARIANT* pOleVariant, OBJECTREF * const & pObj);
+    static void MarshalArrayVariantOleRefToObject(const VARIANT* pOleVariant, OBJECTREF * const & pObj);
+    static void MarshalArrayVariantObjectToOle(OBJECTREF * const & pObj, VARIANT* pOleVariant);
```

Comment:
I would avoid using C++ references (`&`) with `OBJECTREF`. The levels of indirection to these types should be explicit.

Reply:
I'm just being consistent with the callers `MarshalOleVariantForObject` etc. Maybe they can be refactored together?

Reply:
Were they originally `&`?

Reply:
The callers `MarshalOleVariantForObject` and `MarshalObjectForOleVariant` are using `OBJECTREF * const &` in current main.

Reply:
Okay. Let's keep this then.